src/Riok.Mapperly/Descriptors/MappingBuilders/EnumerableMappingBuilder.cs

```diff
@@ -242,49 +210,29 @@ bool elementMappingIsSynthetic
             return null;
 
         var selectMethod = elementMapping.IsSynthetic ? null : ctx.Types.Get(typeof(Enumerable)).GetStaticGenericMethod(SelectMethodName);
-
         return new LinqEnumerableMapping(ctx.Source, ctx.Target, elementMapping, selectMethod, collectMethod);
     }
 
     private static IMethodSymbol? ResolveImmutableCollectMethod(MappingBuilderContext ctx)
     {
-        if (SymbolEqualityComparer.Default.Equals(ctx.Target.OriginalDefinition, ctx.Types.Get(typeof(ImmutableArray<>))))
+        if (ctx.CollectionInfos!.Target.Type == CollectionType.ImmutableArray)
             return ctx.Types.Get(typeof(ImmutableArray)).GetStaticGenericMethod(ToImmutableArrayMethodName);
 
-        if (
-            SymbolEqualityComparer.Default.Equals(ctx.Target.OriginalDefinition, ctx.Types.Get(typeof(ImmutableList<>)))
-            || SymbolEqualityComparer.Default.Equals(ctx.Target.OriginalDefinition, ctx.Types.Get(typeof(IImmutableList<>)))
-        )
+        if (ctx.CollectionInfos!.Target.Type is CollectionType.ImmutableList or CollectionType.IImmutableList)
```

Comment:
Is the null-forgiving operator still needed here? As far as I can see, `ctx.CollectionInfos` cannot be null after the first `if`

Reply:
It isn't, removed.